client/site-profiler/components/site-profiler.tsx

```diff
@@ -64,50 +71,56 @@ export default function SiteProfiler() {
 				</LayoutBlock>
 			) }
 
-			{ siteProfilerData && (
-				<LayoutBlock className="domain-result-block">
-					{
-						// Translators: %s is the domain name searched
-						<DocumentHead title={ translate( '%s ‹ Site Profiler', { args: [ domain ] } ) } />
-					}
-					{ siteProfilerData && (
-						<LayoutBlockSection>
-							<HeadingInformation
-								domain={ domain }
-								conversionAction={ conversionAction }
-								onCheckAnotherSite={ () => updateDomainQueryParam( '' ) }
-								hostingProvider={ hostingProviderData?.hosting_provider }
-								urlData={ urlData }
-							/>
-						</LayoutBlockSection>
-					) }
-					{ ! siteProfilerData.is_domain_available && (
-						<>
-							{ siteProfilerData && (
-								<LayoutBlockSection>
-									<HostingInformation
-										dns={ siteProfilerData.dns }
-										urlData={ urlData }
-										hostingProvider={ hostingProviderData?.hosting_provider }
-									/>
-								</LayoutBlockSection>
-							) }
+			{
+				// For speical vaild domain mapping, we need to wait until the result comes back
+				showResultScreen && (
+					<LayoutBlock className="domain-result-block">
+						{
+							// Translators: %s is the domain name searched
+							<DocumentHead title={ translate( '%s ‹ Site Profiler', { args: [ domain ] } ) } />
+						}
+						{ ( siteProfilerData || specialDomainMapping ) && (
 							<LayoutBlockSection>
-								<DomainInformation
+								<HeadingInformation
 									domain={ domain }
-									whois={ siteProfilerData.whois }
+									conversionAction={ conversionAction }
+									onCheckAnotherSite={ () => updateDomainQueryParam( '' ) }
 									hostingProvider={ hostingProviderData?.hosting_provider }
 									urlData={ urlData }
+									specialDomainMapping={ specialDomainMapping }
 								/>
 							</LayoutBlockSection>
-						</>
-					) }
-				</LayoutBlock>
-			) }
+						) }
+						{ siteProfilerData && ! siteProfilerData.is_domain_available && (
+							<>
+								{ siteProfilerData && (
```

Comment:
We don't need this double check since it is the same check 2 lines above.

Reply:
Nice catch, thanks!